### ISSUE TITLE
Remote kernel with jupyterhub

### ISSUE THREAD

**Issue by kylebarron** [nteract/hydrogen#1274]
I don't see any documentation or other issues talking about using jupyter kernels with jupyterhub. I'm testing out a JupyterHub installation on a server I have access to; when I forward port 8000 to my computer, I can open up a jupyter notebook at `localhost:8000`, but when I try to have Hydrogen connect to port 8000, it doesn't accept my token. 

Is this why the cookie option exists? How do I find the cookie from Jupyterhub?

@nikitakit you added the remote connection, right?

**Comment by nikitakit** [comment]
If your JupyterHub installation is set up to accept tokens, token authentication should be the best way forward.

The cookie option is for the case where JupyterHub has some custom authenticator (e.g. LDAP) that's outside your control; it allows you to connect despite being a gigantic kludge. You can pull out the cookie by logging it to JupyterHub in your browser, opening the browser debug console, and running `document.cookie`. The cookie string can then be pasted into hydrogen (note that the string delimiters shown in the js console are not part of the cookie)

**Comment by andybrnr** [comment]
@kylebarron did you get this sorted out? I'm just beginning to experiment with the goal of a similar kind of deployment. I haven't been able to connect in and launch a new kernel with either the VS Code Jupyter plugin using token auth or with Atom + Hydrogen. Any thoughts on how to approach debugging the failure to get a working connection would be much appreciated.

**Comment by andybrnr** [comment]
Created an issue on the jupyterhub project, but not sure whether it's really a JH issue, a Hydrogen issue, or something in between. I realized I needed to point to a different url when using jupyterhub... e.g. if for a stand alone jupyter server, you point it at base_url/, with jupyterhub you won't hit the proper api endpoint unless pointed at base_url/user/user_name (where the notebook root is after authenticating with PAM). I can authenticate to the server with a token, select a new session (and see the existence of other ones if the kernels are already up), and see the list of available kernels. Clicking on a kernel does start the kernel on the server, which I can see from the JupyterHub logs. However, Hydrogen doesn't seem to realize the kernel is up and running, as the "select kernel" dialog remains presented after kernel start - if I cancel out, pull the connect to remote kernel dialog up again and reauthenticate, I can see the existence of the session for the kernel that I just started, but clicking on the name in the dialog doesn't connect to it. So I guess when running behind jupyterhub, the url paths for the api clearly aren't the same as with the stand alone server - I can't quite figure out what's mismatched, though, as the fact that I can get some dialogs at all means it's talking to the notebook server API, it just can't reach the kernel. Any ideas about where I need to look? Thanks!
